lib/rubocop/ast/node/arg_node.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # A node extension for `arg`, `optarg`, `restarg`, `kwarg`, `kwoptarg`,
+    # `kwrestarg`, `blockarg`, `shadowarg` and `forward_arg` nodes.
+    # This will be used in place of a plain node when the builder constructs
+    # the AST, making its methods available to all `arg` nodes within RuboCop.
+    class ArgNode < Node
+      # Returns the name of an argument.
+      #
+      # @return [Symbol, nil] the name of the argument
+      def name
+        node_parts[0]
+      end
+
+      # Returns the default value of the argument, if any.
+      #
+      # @return [Node, nil] the default value of the argument
+      def default_value
+        return unless optional?
+
+        node_parts[1]
+      end
+
+      # Checks whether the argument is optional (it has a default value)
+      #
+      # @return [Boolean] whether the argument is optional
+      def optional?
```

Comment:
I'm not sure this is the right name, or the right result.
Maybe it should be named `default?`
Or if `optional?`, then I'd expect `argument_list.all?(&:optional?)` to be a good check if there's a mandatory argument, so I'd expect a `true` result for `*x`, `**x`, `...` and `&block`.
Or maybe you want both 😆

Reply:
I was going based on the nodes being `optarg` and `kwoptarg`, but I think you have a valid point. I'll change it to `default?`.